builtin/logical/database/rotation.go

```diff
@@ -324,37 +327,53 @@ func (b *databaseBackend) setStaticAccount(ctx context.Context, s logical.Storag
 		return output, err
 	}
 
-	dbi.RLock()
-	defer dbi.RUnlock()
+	dbi.Lock()
+	defer dbi.Unlock()
```

Comment:
Out of curiosity what was the reasoning for switching to a write lock here?

Reply:
I've reverted this in dec64cd.